packages/pigeon/lib/swift_generator.dart

```diff
@@ -0,0 +1,642 @@
+        indent.writeln('func ${method.name}(${argSignature.join(', ')})');
+      } else {
+        indent.writeln(
+            'func ${method.name}(${argSignature.join(', ')}) -> $returnType');
+      }
+    }
+  });
+
+  indent.addln('');
+  indent.writeln(
+      '/** Generated setup class from Pigeon to handle messages through the `binaryMessenger`.*/');
+  indent.write('class ${apiName}Setup ');
+  indent.scoped('{', '}', () {
+    final String codecName = _getCodecName(prefix, api);
+    indent.format('''
+/** The codec used by $apiName. */
+static var codec: FlutterStandardMessageCodec { $codecName.shared }
+''');
+    indent.writeln(
+        '/** Sets up an instance of `$apiName` to handle messages through the `binaryMessenger`. */');
+    indent.write(
+        'static func setup(binaryMessenger: FlutterBinaryMessenger, api: $apiName) ');
+    // aceitar api null
+    indent.scoped('{', '}', () {
+      for (final Method method in api.methods) {
+        final String channelName = makeChannelName(api, method);
+        final String varChannelName = '${method.name}Channel';
+
+        indent.writeln(
+            'let $varChannelName = FlutterBasicMessageChannel(name: "$channelName", binaryMessenger: binaryMessenger, codec: codec)');
+        indent.write('$varChannelName.setMessageHandler ');
+        indent.scoped('{ message, reply in', '}', () {
+          final List<String> methodArgument = <String>[];
+          if (method.arguments.isNotEmpty) {
+            indent.write('guard let args = message as? [Any?] ');
+            indent.scoped('else {', '}', () {
+              indent.writeln(
+                  'let error = FlutterError(code: "unexpected-args", message: "Unexpected argument parameters", details: nil)');
+              indent.writeln('reply(wrapError(error))');
+              indent.writeln('return');
+            });
+            indent.write('guard args.count == ${method.arguments.length} ');
+            indent.scoped('else {', '}', () {
+              indent.writeln(
+                  'let error = FlutterError(code: "unexpected-args-count", message: "Unexpected argument parameters count", details: "Expected parameters count: ${method.arguments.length}. Received: \\(args.count)")');
+              indent.writeln('reply(wrapError(error))');
+              indent.writeln('return');
+            });
+            enumerate(method.arguments, (int index, NamedType arg) {
+              final String argType = _swiftTypeForDartType(prefix, arg.type);
+              final String argName = _getSafeArgumentName(index, arg);
+
+              indent.write('guard let $argName = args[$index] as? $argType ');
+              indent.scoped('else {', '}', () {
+                indent.writeln(
+                    'let error = FlutterError(code: "unexpected-arg-type", message: "${arg.name} argument unexpected type", details: "Expected type: $argType. Received: \\(type(of: args[$index]))")');
```

Comment:
Similarly, the creation of this error could be a small helper that takes these three variables as parameters, so the string isn't repeated over and over.

And in fact, that will reduce actual binary size, because otherwise every one of these entire strings is duplicated within the binary since they are all slightly different.

Reply:
I got it. I'll review the generator to refactor and avoid those boilerplates.
But, in this case I think similarly above. Instead use a helper, what if remove this?
Is there need to this? Or can we presume that the type is correct either?

Follow some proposes to change in generator.
The force unwrap (`as!`) would be used just in case the type be non-null (`Type`).
To null type (`Type?`), would be used `as?`.

## Arguments parser
This
```swift
let valueArg = args[0] as! Type
```
instead of
```swift
guard let valueArg = args[0] as? Type else {
  let error = FlutterError(code: "unexpected-arg-type", message: "value argument unexpected type", details: "Expected type: Type. Received: \(type(of: args[0]))")
  reply(wrapError(error))
  return
}
```

## .fromMap
This
```swift
let value = map["value"] as! Type
```
instead of
```swift
guard let value = map["value"] as? Type else { return nil }
```

## CodeReader: FlutterStandardReader
This
```swift
return Type.fromMap(self.readValue() as! [String: Any?])  
```
instead of
```swift
guard let map = self.readValue() as? [String: Any?] else {
  return nil
}
return Type.fromMap(map)
```

Reply:
I believe we force-cast on the Dart side, so that's probably reasonable here as well. I'll defer to @gaaclarke on that.

Reply:
I changed it